### PR TITLE
💎(site) Add Sections Title to Stories Landing Page

### PR DESCRIPTION
**Pull request by Vanss472**
- created a `stories-section.njk` file to call stories section
- modified `item-grid` to import `stories-section` if `modifiers.variant == 'stories'`
- ran accessibility fix typo on search box
- changed auto-fit to auto-fill

### CODE REVIEW

File: site/_components/stories-section.njk [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    @param {string} subtitle (optional): Text for a subtitle beneath the primary title.
+    @param {array} items: List of item objects.
+ #}-->
+{% from 'stories-card.njk' import storiesCard %} {% macro storiesSection(titles, items) %} {% for title in titles %}
+<h2 class="item-grid__title type--h2">{{title.title}}</h2>
+{% if title.subtitle %}
+<p class="item-grid__subtitle type--large">{{title.subtitle}}</p>
+{% endif %}
+
+<div class="item-grid__content">{% for item in items %} {% if title.tag == 'pwa' %} {% if item.tag == 'pwa' %} {{storiesCard(item)}} {% endif %} {% endif %} {% if title.tag == 'games' %} {% if item.tag == 'games' %} {{storiesCard(item)}} {% endif %} {% endif %} {% if title.tag == 'android' %} {% if item.tag == 'android' %} {{storiesCard(item)}} {% endif %} {% endif %} {% endfor %}</div>
```
Comment (Snugug) [inline site/_components/stories-section.njk:14]:
```suggestion
<div class="item-grid__content">{% for item in items %} {% if title.tag == item.tag %} {{storiesCard(item)}} {% endif %} {% endfor %}</div>
```

I think this should still work, should clean up this a little bit, and allow for internationalization (for instance, `games` is different in English and Spanish)

File: site/en/_data/stories.yml [hunk 2]
```diff
@@ -2,8 +2,16 @@ featured:
...
+  headings:
+    - tag: pwa
+      title: Progressive Web Applications
+      subtitle:
+    - tag: games
+      title: Games
+      subtitle:
+    - tag: android
+      title: Android
+      subtitle:
```
Comment (Snugug) [inline site/en/_data/stories.yml:14]: This is great! Can you also do the same for Spanish (`es`)? The big difference is that the tag for `games` (and the title) should be `juegos`

File: site/sass/components/_item-grid.scss [hunk 3]
```diff
@@ -130,7 +130,7 @@
       margin-top: 1.5rem;
 
       @include mq(1058px) {
-        grid-template-columns: repeat(auto-fit, minmax(pxRem(400px), 1fr));
+        grid-template-columns: repeat(auto-fill, minmax(pxRem(400px), 1fr));
```
Comment (Snugug) [inline site/sass/components/_item-grid.scss:133]: Why did you swap this over to fill?

File: site/_components/search-box.njk [hunk 4]
```diff
@@ -7,7 +7,7 @@
   @param {object} modifiers - classes that change how to display the search box.
 #}-->
 <div class="search-box {{modifiers.background}} {{modifiers.extended}} {{modifiers.headerClass}}">
-  <button arial-label="open search box" class="search-box__icon">{{ icon('search')}}</button>
+  <button aria-label="open search box" class="search-box__icon">{{ icon('search')}}</button>
```
Comment (Snugug) [inline site/_components/search-box.njk:10]: Great catch! Thanks!

**Review by Snugug — CHANGES_REQUESTED** [review]
Thanks for your work here! Simple solutions are often the best solutions, and who doesn't love more stuff in `item-grid`  😝.

I've got a few comments in-line, but I also noticed a few things that differ from the designs:
* There's less vertical spacing between sections than expected
* The sections don't have their subheading (if you need me to write something, I'm happy to)
* The sections don't have their icon.

Thanks again for your work here!

File: site/en/_data/stories.yml (hunk 2 above)
Comment (Vanss472) [reply site/en/_data/stories.yml:14]: Done.

File: site/en/_data/stories.yml [hunk 5]
```diff
@@ -2,8 +2,16 @@ featured:
   modifiers:
     custom: landing--stories-full-bleed
 sections:
+  headings:
+    - tag: pwa
+      title: Progressive Web Applications
+      subtitle:
```
Comment (Snugug) [inline site/en/_data/stories.yml:8]:
```suggestion
      subtitle: A more powerful web. See how web developers big and small are taking advantage of new web capabilities and Google’s Play Store to deliver next-level user experiences. 
```

File: site/en/_data/stories.yml [hunk 6]
```diff
@@ -2,8 +2,16 @@ featured:
...
+    - tag: games
+      title: Games
+      subtitle:
```
Comment (Snugug) [inline site/en/_data/stories.yml:11]:
```suggestion
      subtitle: Play on! See how companies are optimizing their games for Chrome OS to bring joy to even more people.
```

File: site/en/_data/stories.yml (hunk 2 above)
Comment (Snugug) [inline site/en/_data/stories.yml:14]:
```suggestion
      subtitle: Watch your app grow. Learn how companies are adapting their Android apps to work across millions of large screens and new form factors.
```

**Review by Snugug — CHANGES_REQUESTED** [review]
Just a quick set of 3 with the new subtitles

File: site/sass/components/_item-grid.scss (hunk 3 above)
Comment (Snugug) [reply site/sass/components/_item-grid.scss:133]: Maybe make this something just for the stories modifier for item grid then. I think we in general want `auto-fit` and swapping it here may have unintended consequences elsewhere

Comment (Vanss472) [reply site/sass/components/_item-grid.scss:133]: yup, that's what I did, look ⤵️ 

image

**Comment by Vanss472** [comment]
> I've got a few comments in-line, but I also noticed a few things that differ from the designs:
> 
- [x] * There's less vertical spacing between sections than expected
- [x] * The sections don't have their subheading (if you need me to write something, I'm happy to)
- [x] * The sections don't have their icon.


Fixed.

File: site/sass/components/_item-grid.scss (hunk 3 above)
Comment (Snugug) [reply site/sass/components/_item-grid.scss:140]: Oh I totally missed this! Couldn't tell from how the PR diff worked!

File: site/es/_data/stories.yml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  headings:
+    - tag: pwa
+      title: Progressive Web Applications
+      subtitle:
+    - tag: juegos
+      title: Juegos
+      subtitle:
+    - tag: android
+      title: Android
+      subtitle:
```
Comment (Snugug) [inline site/es/_data/stories.yml:14]:
```suggestion
      subtitle: Mira como crece tu aplicación. Aprende cómo distintas compañías están adaptando sus aplicaciones Android para que funcionen en millones de pantallas grandes y en nuevos tipos de dispositivos.
```

File: site/es/_data/stories.yml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+featured:
+  modifiers:
+    custom: landing--stories-full-bleed
+sections:
+  headings:
+    - tag: pwa
+      title: Progressive Web Applications
+      subtitle:
```
Comment (Snugug) [inline site/es/_data/stories.yml:8]:
```suggestion
      subtitle: Una web poderosa. Mira como desarrolladores web, pequeños y grandes sacan ventaja de las nuevas funcionalidades para la web y para Google Play Store para proveer experiencias de usuario al siguiente nivel.
```

File: site/es/_data/stories.yml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    - tag: juegos
+      title: Juegos
+      subtitle:
```
Comment (Snugug) [inline site/es/_data/stories.yml:11]:
```suggestion
      subtitle: Hora de jugar! Mira cómo  distintas compañías optimizan sus juegos para Chrome OS para brindar diversión a más y más personas.
```

**Review by Snugug — CHANGES_REQUESTED** [review]
Thank you for this! It's _so close_! I think there are just two things (in addition to Spanish subheadings, added below in comments) that I think need to be done:

* The subtitle needs a `max-width` on it. Something like 50 or 60 `ch` should do
* I think a little more logic needs to be added to the sections so they only display if there are stories for it. Now that the new logic is merged in, there aren't any PWAs to display.

Thanks!

**Comment by Vanss472** [comment]
- [x] * The subtitle needs a `max-width` on it. Something like 50 or 60 `ch` should do
 - [x] * I think a little more logic needs to be added to the sections so they only display if there are stories for it. Now that the new logic is merged in, there aren't any PWAs to display.

Done.

**Review by Snugug — APPROVED** [review]

**Comment by Snugug** [comment]
Looks great! Thanks for your work on this!

**Pull request merged**